### PR TITLE
Update 4.2 branch with puppet last changes

### LINKED ISSUE #381: Merge latest changes into 4.2 wazuh-puppet branch

### PR DESCRIPTION
**Pull request by okynos**
Hello team,

This PR closes #381 we want to update last 4.2 changes into puppet 4.2 main branch.
We have done the following tasks:
- Removed `expect`label config of active-response configuration generator.
- Removed proxy option into API configuration (removed from framework).
- Fixed `pdk` validation tests.
- Updated Rubocop file notation and dependencies.
- Updated automated kitchen tests.
- Updated Elasticseach and OpenDistro versions.

Regards.

### CODE REVIEW

File: kitchen/test/integration/agent/agent_spec.rb [hunk 1]
```diff
@@ -15,10 +15,9 @@
 
   # Verifying daemons
   wazuh_daemons = {
-    # 'ossec-agentd' => 'ossec',
-    'ossec-execd' => 'root',
-    # 'ossec-syscheckd' => 'root',
-    # 'wazuh-modulesd' => 'root',
+    'wazuh-agentd' => 'ossec',
```
Comment (alberpilot) [inline kitchen/test/integration/agent/agent_spec.rb:18]: Please complete the daemon list.

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: kitchen/test/integration/agent/agent_spec.rb (hunk 1 above)
Comment (okynos) [reply kitchen/test/integration/agent/agent_spec.rb:18]: Done

**Review by alberpilot — COMMENTED** [review]
LGTM

**Pull request merged**